unlock-app/src/components/interface/Header.js

```diff
@@ -17,18 +17,63 @@ const navigationButtons = [
   Buttons.Telegram,
 ]
 
+const accountTypes = {
+  managed: 'account/managed',
+  crypto: 'account/crypto',
+}
+
+export const typeOfAccount = account => {
+  if (!account) {
+    // this case mostly seems to happen in component tests that include the
+    // header but haven't had an account set.
+    return 'account/undefined'
+  }
+  const { address, emailAddress } = account
+  if (address && emailAddress) {
+    return accountTypes.managed
+  } else if (address) {
+    return accountTypes.crypto
+  } else {
+    // This final catch-all state is invalid. No app navigation will display for
+    // a user who doesn't match the above conditions. A user may be in an
+    // intermediate state (not all account updates have hit redux), which may
+    // cause a brief flicker of the icons. The return value here should indicate
+    // what pieces are missing for debugging purposes.
+    return `account/address={${address}}--emailAddress={${emailAddress}}`
```

Comment:
Not a blocker from shipping this, we may need to do some research for understand the scenarios that have users that end up here and chip away at removing them

Reply:
Agreed 100%